pkg/reconciler/v1alpha1/revision/resources/deploy.go

```diff
@@ -158,6 +150,28 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, observab
 	return podSpec
 }
 
+func getUserPort(rev *v1alpha1.Revision) int32 {
```

Comment:
could we have this return int and the following two functions consuming ints instead of int32... may save some conversions

Reply:
It is pulled from rev.Spec.Container.Ports as an int32

It is needed in the Deployment as:
* int32 for Container.Ports
* IntOrString for Probes
* string for Env

I may keep this the same. I will look at converting to string once in makePodSpec and then update UserProbe and Env to take a String.